app/services/okta.rb

```diff
@@ -0,0 +1,71 @@
+# frozen_string_literal: true
+
+class Okta
+  include HTTParty
+  LEEWAY = 0
+  base_uri ENV.fetch("OKTA_SERVER_URL")
+
+  class << self
+    def find_user_by_id_token(id_token)
+      info = validate_okta_id_token(id_token)
+      find_user(info) || create_user(info)
+    rescue JWT::DecodeError => e
+      raise Okta::FailedAuthentication, e.message
+    end
+
+    private
+
+    def find_user(user_info)
+      User.find_by(sso_guid: user_info[:sso_guid])
+    end
+
+    def create_user(user_info)
+      User.create!(user_info)
+    end
+
+    def validate_okta_id_token(id_token)
+      payload = transform_jwt_payload JWT.decode(id_token, nil, true, jwt_options).first
+      raise "ID Token does not include sso guid, make sure login scope includes profile" unless payload[:sso_guid]
```

Comment:
I'm having trouble getting the `id_token` to actually include the sso guid, or any other profile claims beyond the minimal claims.

from the Okta [documentation](https://developer.okta.com/docs/reference/api/oidc/#scope-dependent-claims-not-always-returned)

> Scope-dependent claims are returned differently depending on the values in response_type and the scopes requested

We are using the recommended SPA authorization code flow, which doesn't appear to be including any custom profile attributes in the `id_token`. I was able to fetch all the attributes including `ssoGuid` using the corresponding `access_token` on the `/userinfo` endpoint

Reply:
this comment is no longer relevant since Spencer is switching to the `access_token`